api/internal/localizer/localizer.go

```diff
@@ -368,6 +425,91 @@ func (lc *localizer) localizeDir(path string) (string, error) {
 	return locPath, nil
 }
 
+// copyChartHome copies helm chart home path, relative to root that lc is at, to
+// the same location relative to dst and returns said relative copied location.
+//
+// If path indicates the default chart home, copyChartHome copies chart home to
+// the same location in dst, without following symlinks.
+func (lc *localizer) copyChartHome(path string) (string, error) {
+	if filepath.IsAbs(path) {
+		return "", errors.Errorf("absolute path %q not handled in alpha", path)
+	}
+	locPath, err := filepath.Rel(lc.root.String(), lc.root.Join(path))
+	if err != nil {
+		return "", errors.WrapPrefixf(err, "no path from root to chart home")
+	}
+	if path == "" {
+		path = types.HelmDefaultHome
+		locPath = ""
+	}
+	// chart home may serve as untar destination
+	if !lc.fSys.Exists(lc.root.Join(path)) {
+		return locPath, nil
+	}
+	// perform localize directory checks
+	ldr, err := lc.ldr.New(path)
+	if err != nil {
+		return "", errors.Wrap(err)
+	}
+	clean, err := filesys.ConfirmDir(lc.fSys, ldr.Root())
+	if err != nil {
+		log.Panicf("unable to establish validated directory %q: %s", ldr.Root(), err)
+	}
+	var copyDst string
+	isDefault := lc.root.Join(path) == lc.root.Join(types.HelmDefaultHome)
+	if isDefault {
+		copyDst = types.HelmDefaultHome
+		if lc.copiedHelmDefaultHome {
+			return locPath, nil
```

Comment:
I find the structure of the default chart handling home here a bit messy and convoluted. When we structure code, we should try to return early where we can, minimize code, and have related logic together. I'm curious why we need to have this separate `isDefault` bool? It seems like we are spreading the logic of default-home handling to various parts of the code when we don't need to. I think it would make more sense to do something like

```
func (lc *localizer) copyChartHome(path string) (string, error) {
  if path == "" {
    if lc.copiedHelmDefaultHome {
			return "", nil // return early here if we can, instead of executing all the other code
    } 
    path = types.HelmDefaultHome
    lc.copiedHelmDefaultHome = true
  } 
 .... // everything else
}
```

I'm having trouble understanding why later in the function, there are different branches whether we are using the default "charts" or some other passed-in value. The perspective of this function should be to treat "" as if it was passed in "charts", so why does the rest of the function care whether the value came from a default or not?

Reply:
The `isDefault` logic is here to determine the copy destination of the chart home. The cases are:
* Chart homes that are not the default are copied to their cleaned locations, devoid of symlinks, like other files in the localize destination. 
* The default chart home must be copied to `charts`, regardless of the actual cleaned location of the directory. 

Assuming we mind copying `charts` multiple times, the default chart home is not only when we pass in `""` as the path, but also any form of `charts`, `./charts`.

Let me try to improve the readability of this function.

Reply:
I've split the original `copyChartHome` into 2 functions. Hopefully it's more readable now.